python/ray/util/spark/cluster_init.py

```diff
@@ -211,6 +217,11 @@ def shutdown(self, cancel_background_job=True):
         if not self.is_shutdown:
             self.disconnect()
             os.environ.pop("RAY_ADDRESS", None)
+
+            if self.global_cluster_lock_fd is not None:
+                # release global mode cluster lock.
+                fcntl.flock(self.global_cluster_lock_fd, fcntl.LOCK_UN)
```

Comment:
How do we invoke `shutdown_ray_cluster` to shutdown the global ray cluster in a different notebook when `global _active_ray_cluster` doesn't really exist?

Reply:
> How do we invoke shutdown_ray_cluster to shutdown the global ray cluster

I am not going to support :) It makes things complicated

Reply:
Agreed. This could be a highly disruptive experience for users (user A running a 10 hour job, user B attaches to the cluster and submits a 5 minute job and calls shutdown... 9.5 hours into user A's job. They might get a bit angry) and would make this difficult to use from a process / coordination point of view.